webrender/src/renderer.rs

```diff
@@ -2377,7 +2377,15 @@ impl Renderer {
                     // Add a new document to the active set, expressed as a `Vec` in order
                     // to re-order based on `DocumentLayer` during rendering.
                     match self.active_documents.iter().position(|&(id, _)| id == document_id) {
-                        Some(pos) => self.active_documents[pos].1 = doc,
+                        Some(pos) => {
+                            // If the document we are replacing must be drawn
+                            // (in order to update the texture cache), issue
+                            // a render just to off-screen targets.
+                            if self.active_documents[pos].1.frame.must_be_drawn() {
+                                self.render_impl(None).ok();
```

Comment:
technically, we only need to render the `pos` indexed document, not all of them.
Perhaps, we could accumulate the documents into something like `inactive_documents` vector that is flushed whenever we happen to render stuff?

Reply:
True - would it be OK to do that as a follow up?